### PR TITLE
fix: re-check ongoing/unreleased items

### PR DESCRIPTION
**Pull request by dreulavelle**
# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

## Summary by CodeRabbit

- **New Features**
	- Improved functionality for managing ongoing and unreleased items.
	- Enhanced logging for better visibility of state updates.
	- Updated state management for media items to ensure consistent state storage across hierarchy.

- **Bug Fixes**
	- Simplified query logic for item state updates, removing unnecessary conditions.

- **Refactor**
	- Renamed method to better reflect its purpose and updated scheduling accordingly.
	- Updated method signatures to improve clarity on return types, now returning state tuples.

### CODE REVIEW

File: src/program/program.py [hunk 1]
```diff
@@ -212,39 +212,43 @@ def _retry_library(self) -> None:
...
                 select(MediaItem.id)
                 .where(MediaItem.type.in_(["movie", "show"]))
                 .where(MediaItem.last_state.in_([States.Ongoing, States.Unreleased]))
-                .where(or_(MediaItem.aired_at <= datetime.now(), MediaItem.aired_at.is_(None)))
             ).scalars().all()
 
-            if len(results) == 0:
-                logger.log("PROGRAM", "No ongoing or unreleased items to reindex.")
+            if not item_ids:
+                logger.log("PROGRAM", "No ongoing or unreleased items to update.")
```
Comment (Gaisberg) [inline src/program/program.py:225]: Debug log maybe?

File: src/program/program.py [hunk 2]
```diff
@@ -212,39 +212,43 @@ def _retry_library(self) -> None:
...
+            for item_id in item_ids:
                 try:
                     item = session.execute(select(MediaItem).filter_by(id=item_id)).unique().scalar_one_or_none()
                     if item:
-                        for indexed_item in TraktIndexer().run(item):
-                            indexed_item.store_state()
-                            session.merge(indexed_item)
-                            logger.debug(f"Reindexed {indexed_item.log_string} ({indexed_item.id})")
+                        previous_state = item.last_state
+                        item.store_state()
```
Comment (Gaisberg) [inline src/program/program.py:236]: Could store_state() return previous, new state tuple instead?

File: src/program/program.py [hunk 3]
```diff
@@ -212,39 +212,43 @@ def _retry_library(self) -> None:
...
+                        if previous_state != item.last_state:
+                            self.em.add_event(Event(emitted_by="UpdateOngoing", item_id=item_id))
```
Comment (Gaisberg) [inline src/program/program.py:238]: This will not add shows with only episodes changed into queue

File: src/program/program.py [hunk 4]
```diff
@@ -212,39 +212,43 @@ def _retry_library(self) -> None:
...
+                            logger.debug(f"Updated state for {item.log_string} ({item.id}) from {previous_state} to {item.last_state}")
+                            counter += 1
+                        session.merge(item)
                         session.commit()
                 except Exception as e:
-                    logger.error(f"Failed to reindex item with ID {item_id}: {e}")
-            logger.log("PROGRAM", "Reindexing completed.")
+                    logger.error(f"Failed to update state for item with ID {item_id}: {e}")
+
+            logger.log("PROGRAM", f"Found {counter} items with updated state.")
```
Comment (Gaisberg) [inline src/program/program.py:246]: Again this should be debug

**Review by Gaisberg — CHANGES_REQUESTED** [review]

File: src/program/program.py (hunk 2 above)
Comment (dreulavelle) [reply src/program/program.py:236]: that could be useful

File: src/program/media/item.py [hunk 5]
```diff
@@ -141,11 +141,13 @@ def __generate_composite_key(item: dict) -> str | None:
...
         return f"{item_type}_{trakt_id}"
 
-    def store_state(self, given_state=None) -> None:
+    def store_state(self, given_state=None) -> tuple[States, States]:
+        """Store the state of the item."""
         new_state = given_state if given_state else self._determine_state()
         if self.last_state and self.last_state != new_state:
             sse_manager.publish_event("item_update", {"last_state": self.last_state, "new_state": new_state, "item_id": self.id})
         self.last_state = new_state
+        return (self.last_state, new_state)
```
Comment (Gaisberg) [inline src/program/media/item.py:150]: This wont do, as last_state is written in the line before. Could you store last_state into a temporary variable and provide it here?

**Review by Gaisberg — CHANGES_REQUESTED** [review]

File: src/program/media/item.py (hunk 5 above)
Comment (dreulavelle) [reply src/program/media/item.py:150]: oh thats true, my bad!

**Review by Gaisberg — APPROVED** [review]

**Pull request merged**